Wrong PRIVATE attribute for a USE-associated symbol

The code causes an error because F_Front generates a wrong PRIVATE statement.

```
$ F_Front hoge.f90|F_Decomp > fff.f90
$ gfortran fff.f90 -c
fff.f90:12:20: Error: Cannot change attributes of USE-associated symbol yyy at (1)
fff.f90:13:5:

  INTERFACE xxx
     1
Error: Expecting END INTERFACE statement at (1)
```

```fortran
module mmm
  private
  INTERFACE xxx
     SUBROUTINE yyy
     END SUBROUTINE yyy
  END INTERFACE xxx
end module mmm

subroutine sub
  use mmm
  INTERFACE xxx
     SUBROUTINE yyy
     END SUBROUTINE yyy
  END INTERFACE xxx
end subroutine sub
```

--- a/F-BackEnd/src/xcodeml/f/decompile/XfDecompileDomVisitor.java
+++ b/F-BackEnd/src/xcodeml/f/decompile/XfDecompileDomVisitor.java
@@ -7463,7 +7463,11 @@ XfDecompileDomVisitor {
             } else {
                 NodeList list = n.getChildNodes();
                 for (int i = 0; i < list.getLength(); i++) {
-                    this.enter(list.item(i));
+                    Node child = list.item(i);
+                    if ("FinterfaceDecl".equals(child.getNodeName())) {
+                        continue;
+                    }
+                    this.enter(child);
                 }
             }
         }

--- a/F-FrontEnd/test/testdata/interface_private.f90
+++ b/F-FrontEnd/test/testdata/interface_private.f90
@@ -1,0 +1,15 @@
+module mmm
+  private
+  INTERFACE xxx
+     SUBROUTINE yyy
+     END SUBROUTINE yyy
+  END INTERFACE xxx
+end module mmm
+
+subroutine sub
+  use mmm
+  INTERFACE xxx
+     SUBROUTINE yyy
+     END SUBROUTINE yyy
+  END INTERFACE xxx
+end subroutine sub

